pkg/kubelet/cm/cpumanager/cpu_manager_test.go

```diff
@@ -40,6 +40,12 @@ import (
 	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager"
 )
 
+type mockSourcesReady struct{}
+
+func (s *mockSourcesReady) AddSource(source string) {}
+
+func (s *mockSourcesReady) AllReady() bool          { return false }
+
```

Comment:
There already exists a `sourcesReadyStub` you can use here instead of this.

Reply:
Removing this should fix your `gofmt` error in the `pull-kubernetes-verify` test as well.

Reply:
sourcesReadyStub.AllReady() returns "true", which causes the new call to removeStaleState() to try to actually do stuff, which causes all sorts of grief

Reply:
I see. I'd rather not artifically turn off valid code paths that might uncover other underlying problems that are lurking though. 

Below is a diff to your current patch that deals with the issue you are seeing in a way that's more consistent with the rest of the tests in this file:

```
diff --git a/pkg/kubelet/cm/cpumanager/cpu_manager_test.go b/pkg/kubelet/cm/cpumanager/cpu_manager_test.go
index 7a6724d76c2..bd18b2c95eb 100644
--- a/pkg/kubelet/cm/cpumanager/cpu_manager_test.go
+++ b/pkg/kubelet/cm/cpumanager/cpu_manager_test.go
@@ -40,12 +40,6 @@ import (
 	"k8s.io/kubernetes/pkg/kubelet/cm/topologymanager"
 )

-type mockSourcesReady struct{}
-
-func (s *mockSourcesReady) AddSource(source string) {}
-
-func (s *mockSourcesReady) AllReady() bool { return false }
-
 type mockState struct {
 	assignments   state.ContainerCPUAssignments
 	defaultCPUSet cpuset.CPUSet
@@ -275,14 +269,14 @@ func TestCPUManagerAdd(t *testing.T) {
 				err: testCase.updateErr,
 			},
 			containerMap:      containermap.NewContainerMap(),
-			activePods:        func() []*v1.Pod { return nil },
 			podStatusProvider: mockPodStatusProvider{},
+			sourcesReady:      &sourcesReadyStub{},
 		}

-		mgr.sourcesReady = &mockSourcesReady{}
-
 		pod := makePod("fakePod", "fakeContainer", "2", "2")
 		container := &pod.Spec.Containers[0]
+		mgr.activePods = func() []*v1.Pod { return []*v1.Pod{pod} }
+
 		err := mgr.Allocate(pod, container)
 		if !reflect.DeepEqual(err, testCase.expAllocateErr) {
 			t.Errorf("CPU Manager Allocate() error (%v). expected error: %v but got: %v",
@@ -495,12 +489,13 @@ func TestCPUManagerAddWithInitContainers(t *testing.T) {
 			state:             state,
 			containerRuntime:  mockRuntimeService{},
 			containerMap:      containermap.NewContainerMap(),
-			activePods:        func() []*v1.Pod { return nil },
 			podStatusProvider: mockPodStatusProvider{},
+			sourcesReady:      &sourcesReadyStub{},
+			activePods: func() []*v1.Pod {
+				return []*v1.Pod{testCase.pod}
+			},
 		}

-		mgr.sourcesReady = &mockSourcesReady{}
-
 		containers := append(
 			testCase.pod.Spec.InitContainers,
 			testCase.pod.Spec.Containers...)
@@ -1031,14 +1026,14 @@ func TestCPUManagerAddWithResvList(t *testing.T) {
 				err: testCase.updateErr,
 			},
 			containerMap:      containermap.NewContainerMap(),
-			activePods:        func() []*v1.Pod { return nil },
 			podStatusProvider: mockPodStatusProvider{},
+			sourcesReady:      &sourcesReadyStub{},
 		}

-		mgr.sourcesReady = &mockSourcesReady{}
-
 		pod := makePod("fakePod", "fakeContainer", "2", "2")
 		container := &pod.Spec.Containers[0]
+		mgr.activePods = func() []*v1.Pod { return []*v1.Pod{pod} }
+
 		err := mgr.Allocate(pod, container)
 		if !reflect.DeepEqual(err, testCase.expAllocateErr) {
 			t.Errorf("CPU Manager Allocate() error (%v). expected error: %v but got: %v",
```

Reply:
Thanks for the test changes, applied.